.github/workflows/code_quality.yml

```diff
@@ -0,0 +1,52 @@
+name: 🚧 Lint and run test suite
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  lint:
+    name: 🔎 Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: set up node env
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18.x
+      - name: Install dependencies
+        run: npm ci
+      - name: Run lint
+        run: npm run lint
```

Comment:
The lint script also tries to fix what it can (because of the '--fix` flag in the command) and this changes the files. This should still be safe; any changed files simply disappear when the runner disappears. We're also not interested in these changes- we're just checking for any errors. 

But if we do start having trouble with this job in the future, we should keep this behavior in mind